### ISSUE TITLE
TypeError: replace_refs_factory() missing 1 required positional argument: 'target'

### ISSUE THREAD

**Issue by tomduck** [tomduck/pandoc-fignos#46]
The above error with pandoc-fignos was reported at [greenelab/deep-review](https://github.com/greenelab/deep-review/pull/774#issuecomment-357519950):

**The problem:** There is an API incompatibility between pandoc-fignos 1.0.0 and pandoc-xnos 0.15 (the library for the pandoc-xnos/eqnos/tablenos filters).

**The solution:** Upgrade to the most recent version of pandoc-fignos.

I would like to prevent such errors from occurring in the future.  One way forward is to move pandoc-xnos to MAJOR.MINOR.PATCH versioning and require a particular MAJOR.MINOR version for a given pandoc-fignos release.  The MINOR number would be incremented for API changes.

Feedback is welcome.

**Comment by dhimmel** [comment]
> The MINOR number would be incremented for API changes

Do you mean MAJOR or MINOR. My understanding is that you're proposing that breaking API changes in the pandoc-xnos architecture would increment the MAJOR version?

In other words, the idea here is to make sure fignos and xnos are both using the same MAJOR, or the same MAJOR.MINOR?

**Comment by tomduck** [comment]
My mistake. It is the MAJOR version number that I should be changing in pandoc-xnos when there are API changes.  Pandoc-fignos will require a particular MAJOR version number of the pandoc-xnos library.  This will be written into the pandoc-fignos setup, so it should be completely transparent to users.  Thanks for the feedback.

**Comment by dhimmel** [comment]
> It is the MAJOR version number that I should be changing in pandoc-xnos when there are API changes.

Got it! Makes sense to me.

**Comment by tomduck** [comment]
pandoc-xnos 1.0.1 is now released, and the requirement that pandoc-fignos use pandoc-xnos>=1.0.1,<2 is published in the `nextrelease` branch.

I don't think that publishing a new release is urgent, but users should let me know if they disagree.  Otherwise the changes will go out when I have finished some other work in the queue.  I also want to keep this Issue open in case others encounter the same problem.
